### PR TITLE
Remove legacy module

### PR DESCRIPTION
**Pull request by olijeffers0n**
Removes the legacy module in favour for reflection, not tested.

If they are using legacy they can deal with the bad performance of reflection

### CODE REVIEW

File: pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java [hunk 1]
```diff
@@ -27,7 +33,16 @@ public long getChunkKey(final int x, final int z) {
      * Get the block type from a chunk snapshot at the given coordinates
      */
     public Material getMaterial(ChunkSnapshot snapshot, int x, int y, int z) {
-        return BLOCK_PARSER.getBlockMaterialAt(snapshot, x, y, z);
+        if (BukkitVersionUtil.isUnder(13)) {
+            if (materialMethod == null || blockTypeMethod == null) return null;
```
Comment (bsommerfeld) [inline pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java]: you are double-checking here if the version is < 13

File: pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java [hunk 2]
```diff
@@ -27,7 +33,16 @@ public long getChunkKey(final int x, final int z) {
...
+            try {
+                return (Material) materialMethod.invoke(null, blockTypeMethod.invoke(snapshot, x, y, z));
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                e.printStackTrace();
+                return null;
+            }
+        } else
```
Comment (bsommerfeld) [inline pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java]: add brackets here since the if statement before has some

**Review by bsommerfeld — CHANGES_REQUESTED** [review]

File: pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java (hunk 1 above)
Comment (olijeffers0n) [reply pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java]: I mean, yes. But at the same time I don't want to use the nullability check as a version check

**Comment by olijeffers0n** [comment]
Also, I feel it is important to stress that this is untested and it possible it will not work. Do not merge, just approve

File: pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java (hunk 1 above)
Comment (bsommerfeld) [reply pathetic-model/src/main/java/org/patheloper/util/ChunkUtils.java]: you could throw an illegalstate if one of them is null instead of returning null

**Comment by GodCipher** [comment]
even if, it would "just" destroy our trunk without further impact

**Comment by bsommerfeld** [comment]
> Also, I feel it is important to stress that this is untested and it possible it will not work. Do not merge, just approve

even if, it would "just" destroy our trunk without further impact

**Comment by bsommerfeld** [comment]
but stress is a good keyword. since you mentioned "bad performance", this is worth a stress testing

**Comment by olijeffers0n** [comment]
The actions are messed up cause there isnt a master branch. GH actions only supports workflow "chaining" on master/main branches. Will test later

**Comment by bsommerfeld** [comment]
but why are those commits in this branch?

**Review by bsommerfeld — APPROVED** [review]

**Pull request merged**